cdk/domino_cdk/eks_stack.py

```diff
@@ -400,17 +402,78 @@ def provision_eks_iam_policies(self):
             managed_policies=managed_policies,
         )
 
+    def _get_machine_image(self, cfg_name: str, cfg: Dict[str, Any]) -> Tuple[Optional[str], Optional[str]]:
+        ami_id = cfg.get("ami_id")
+        user_data = cfg.get("user_data")
+        if ami_id and not user_data:
+            raise Exception("User data must be provided when specifying a custom AMI")
+        if ami_id and (cfg.get("ssm_agent") or cfg.get("labels") or cfg.get("taints")):
+            raise Exception(
+                "ssm_agent, labels and taints cannot be automatically configured when specifying a custom AMI. "
+                "You need to configure all of this using user_data."
+            )
+        return ami_id, user_data
+
+    def _handle_user_data(
+        self, name: str, custom_ami: bool, cfg: dict, user_data_list: List[Union[ec2.UserData, str]]
+    ) -> Optional[ec2.UserData]:
+        proceed: bool = False
+        mime_user_data = ec2.MultipartUserData()
+
+        # If we are using default EKS image, tweak kubelet
+        if not custom_ami:
+            mime_user_data.add_part(
+                ec2.MultipartBody.from_user_data(
+                    ec2.UserData.custom(
+                        'KUBELET_CONFIG=/etc/kubernetes/kubelet/kubelet-config.json\n'
+                        'echo "$(jq \'.eventRecordQPS=0\' $KUBELET_CONFIG)" > $KUBELET_CONFIG'
```

Comment:
This happens before kubelet setup. Before bootstrap.sh.

Why. Because bootstrap.sh or whatever does start kubelet. But kubelet is installed in the image and kubelet_config.json does exist. So in this script we tweak existing kubelet_config.json then we let bootstrap to start kubelet with any kubelet_extra_args.

Reply:
EKS adds its logic as LAST part.

Reply:
Got it